### DISCUSSION TITLE
Improve S3 image storing - Do NOT tell people to make their buckets public

### DISCUSSION THREAD

**Discussion by ShivamJoker** [medusajs/medusa#3938] [Feature Requests]
I want to have the option to specify a image CDN like Cloudfront, Imagekit or Cloudflare Images.

These providers can access S3 images, cache them and serve them without making them public.

Here is what needs to be done:
1. The plugins job should only to store the images on S3
2. Update the path in the DB
3. Have an option in Medusa config to specify the CDN base URL (It can be changed, depending on what user wants to use)
4. Update the AWS SDK to v3

I am happy to help and work together on this with the core team.

Thanks for work you guys have done.

**Comment by SGFGOV** [comment]
Just a thought...why not just extend the image entity to save an alternative cdn path

Updatetge retrieve method to return thr can path if it's not null

**Reply by ShivamJoker** [reply]
it should be getting saved every time.
